### DISCUSSION TITLE
Long-press should cycle through checkmark states and prevent immediate reordering

### DISCUSSION THREAD

**Discussion by hiqua** [iSoron/uhabits#902] [Feature Requests]
If you sort habits e.g. by score, ticking one habit might change its relative order.

In v2, there are more than two states, and you possibly want to tick the habit several times to cycle through these states. The reordering makes it annoying, since you have to scroll to where the habit moved to.

We should allow the user to keep pressing so that the checkmarks cycles through possible states (ticked, skipped, unticked, missed or something like that). There should be some visual feedback (like a toast) other than the checkmark itself, since it won't be visible under the user's finger.

Alternative: long-press could prompt the user for what they want: skip, tick, etc.

**Comment by iSoron** [comment]
I agree that this is an issue. Another solution is to keep the current toggling UI unchanged, but only resort the list a few seconds (2-3 seconds) after the user has toggled, not immediately. This would allow the user to toggle it twice before the habit moves somewhere else.

**Reply by lukaszlew** [reply]
Another problem is that if one has 'hide completed' option set, there is no way to mark skip day.
I think 3 seconds delay would be ideal here.

**Reply by raphtastisch** [reply]
A short delay would be absolute perfect to mark skip days with "hide completed" activated.

**Comment by DeWarner** [comment]
#1207 could be another way of solving this issue
